src/styles/calendar.scss

```diff
@@ -0,0 +1,63 @@
+.calendar {
+  width: 730px;
+  border: 2px solid black;
+  padding: 30px 10px;
+
+  display: flex;
+  align-items: center;
+  flex-wrap: wrap;
+  gap: 1px;
+
+  @each $day-name, $day in $start-days {
+    &--start-day-#{$day-name} {
+      .calendar__day:nth-child(1) {
+        margin-left: 100px * $day + $day;
+      }
+    }
+  }
+
+  @for $day from 28 to $month-length + 1 {
+    &--start-day-#{$day} {
+      @for $day-num from 1 to $day + 1 {
+        .calendar__day:nth-child(#{$day-num}) {
+          display: block;
```

Comment:
I think it's better to add display: none only to the last elements that don't match the modifier